src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/Logging/LoggerExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.Logging;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.Telemetry.Logging;
+
+/// <summary>
+/// Extensions for <see cref="ILogger"/>.
+/// </summary>
+[Experimental]
+public static class LoggerExtensions
+{
+    /// <summary>
+    /// Emits a structured log entry.
+    /// </summary>
+    /// <param name="logger">The logger to emit the log entry to.</param>
+    /// <param name="logLevel">Entry will be written on this level.</param>
+    /// <param name="eventId">Id of the event.</param>
+    /// <param name="properties">The set of name/value pairs to log with this entry.</param>
+    public static void StructuredLog(this ILogger logger, LogLevel logLevel, EventId eventId, LoggerMessageProperties properties)
+    {
+        Throw.IfNull(logger).Log<LoggerMessageProperties>(logLevel, eventId, properties, null, (_, _) => string.Empty);
```

Comment:
There's no formatted message? What does the console logger do with this?

Reply:
We could do two things:

* Add a formatter than cranks up a string builder and concats all the properties together in some legible form.

* Extend the console logger to support structured logging by outputting dimensions in a first class way. This was on the todo list but was punted. I suggest we bring this back since it's a fairly easy drop-in which enables more efficient logging use, and supports enrichment and redaction cleanly.

Reply:
> Add a formatter than cranks up a string builder and concats all the properties together in some legible form.

I think this is what we should do for now since it'll work for most logging providers (they all call the formatter by default).

Reply:
> Extend the console logger to support structured logging by outputting dimensions in a first class way. This was on the todo list but was punted. I suggest we bring this back since it's a fairly easy drop-in which enables more efficient logging use, and supports enrichment and redaction cleanly.

Do we have this issue filed?

Reply:
I did have an issue filed to move the R9 console logging functionality into .NET 8. But we ran out of time on that (API review overhead, etc) so decided for now to try and move it to the Otel console logging so at least the logic would be out in the wild for folks to adopt.

However, I think we should try to pursue something more modest. Rather than try to get all of the R9 features into the existing console logger, we should try to just get support for dimensions. It ends up just being a for loop somewhere that iterates through the name/value pairs and maybe an option to control whether this happens or not.

Reply:
We should revive the issue on dotnet/runtime regardless.